### PR TITLE
[java] Fix 15634/ensure driver closed java

### LINKED ISSUE #15634: [🐛 Bug]: Ensuring Drivers close
### Description

Selenium closes the driver as part of the quit method, but it is possible for Selenium to start the driver, but error when creating the session, which prevents the driver process from being closed. 

I see this for sure in Ruby, but I think this applies to other languages as well, so we should check. 

### Reproducible Code

```shell
n/a
```

### Debugging Logs

```logs
n/a
```

### PR DESCRIPTION
**Pull request by iampopovich**
### 🔗 Related Issues
partially fixes #15634 for java

### 💥 What does this PR do?
similar changes as for python in #15636 

### 🔧 Implementation Notes
This pull request enhances the robustness of driver service management in Selenium by ensuring services are properly stopped when session creation fails. It also introduces unit tests across multiple browser drivers to validate this behavior. The most important changes include updates to `RemoteWebDriver` for cleanup logic and new tests for driver service cleanup across Chrome, Edge, Firefox, and Safari.

### Driver Service Cleanup Enhancements:
* [`java/src/org/openqa/selenium/remote/RemoteWebDriver.java`](diffhunk://#diff-0cf141b01e48a733ec9037c51d8e610e7727d05a6b79aa0176def7c3a0fc311bR245): Added logic to stop the driver service if session creation fails, preventing zombie processes. This includes a `try-catch` block in the `startSession` method and the use of `DriverCommandExecutor` for cleanup. [[1]](diffhunk://#diff-0cf141b01e48a733ec9037c51d8e610e7727d05a6b79aa0176def7c3a0fc311bR245) [[2]](diffhunk://#diff-0cf141b01e48a733ec9037c51d8e610e7727d05a6b79aa0176def7c3a0fc311bR286-R296)

### Unit Tests for Driver Service Cleanup:
* [`java/test/org/openqa/selenium/chrome/ChromeDriverServiceCleanupTest.java`](diffhunk://#diff-c16ee27dd948f468311ed0d6a31ed53b9ca53ee67c2ac43d811e1d22de9a2eecR1-R48): Added a test to verify that the Chrome driver service stops when session creation fails due to invalid options.
* [`java/test/org/openqa/selenium/edge/EdgeDriverServiceTest.java`](diffhunk://#diff-c1ffb6392396bd8f50de6afc92b2194f7b7d27fdc8f2ef683690fdb1545f4dc1R92-R109): Added a test to ensure the Edge driver service stops when session creation fails due to invalid options.
* [`java/test/org/openqa/selenium/firefox/GeckoDriverServiceTest.java`](diffhunk://#diff-7e9691bee739620eae799ae2ee7af207080ecba0716c58e35c9300dfff858f50R52-R69): Added a test to confirm that the Firefox driver service stops when session creation fails due to an invalid binary path.
* [`java/test/org/openqa/selenium/safari/SafariDriverServiceTest.java`](diffhunk://#diff-c0bb73b3c1e76a971e4877983cdf5de92349483293cc37a472641493c7c2146eR53-R70): Added a test to verify that the Safari driver service stops when session creation fails due to invalid capabilities.

### 🔄 Types of changes

- Bug fix (backwards compatible)

___

### **PR Type**
Bug fix, Tests

___

### **Description**
- Fix driver service cleanup when session creation fails

- Add unit tests for service cleanup across Chrome, Edge, Firefox, Safari

- Prevent zombie processes by stopping services on session failure

- Enhance error handling in RemoteWebDriver session initialization

___

### **Changes diagram**

```mermaid
flowchart LR
  A["Session Creation"] --> B["Try Block"]
  B --> C["Execute NEW_SESSION"]
  C --> D["Success?"]
  D -->|Yes| E["Set Capabilities & SessionId"]
  D -->|No| F["Catch Exception"]
  F --> G["Check DriverCommandExecutor"]
  G --> H["Stop Service"]
  H --> I["Throw Original Exception"]
```

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>RemoteWebDriver.java</strong><dd><code>Add session failure cleanup logic</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/remote/RemoteWebDriver.java

<li>Wrap session creation in try-catch block<br> <li> Add service cleanup logic when session fails<br> <li> Import DriverCommandExecutor for cleanup operations<br> <li> Prevent zombie processes by stopping services

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16038/files#diff-0cf141b01e48a733ec9037c51d8e610e7727d05a6b79aa0176def7c3a0fc311b">+46/-33</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ChromeDriverServiceCleanupTest.java</strong><dd><code>Add Chrome service cleanup test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/chrome/ChromeDriverServiceCleanupTest.java

<li>New test class for Chrome driver service cleanup<br> <li> Test service stops when session creation fails<br> <li> Use invalid user-data-dir to trigger failure<br> <li> Verify service is not running after failure

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16038/files#diff-c16ee27dd948f468311ed0d6a31ed53b9ca53ee67c2ac43d811e1d22de9a2eec">+48/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>EdgeDriverServiceTest.java</strong><dd><code>Add Edge service cleanup test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/edge/EdgeDriverServiceTest.java

<li>Add new test method for service cleanup<br> <li> Import additional assertion methods<br> <li> Test Edge driver service stops on failure<br> <li> Use invalid user-data-dir argument

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16038/files#diff-c1ffb6392396bd8f50de6afc92b2194f7b7d27fdc8f2ef683690fdb1545f4dc1">+20/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>GeckoDriverServiceTest.java</strong><dd><code>Add Firefox service cleanup test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/firefox/GeckoDriverServiceTest.java

<li>Add service cleanup test for Firefox<br> <li> Import assertion methods for testing<br> <li> Use invalid binary path to trigger failure<br> <li> Verify service stops after session failure

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16038/files#diff-7e9691bee739620eae799ae2ee7af207080ecba0716c58e35c9300dfff858f50">+21/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>SafariDriverServiceTest.java</strong><dd><code>Add Safari service cleanup test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/test/org/openqa/selenium/safari/SafariDriverServiceTest.java

<li>Add service cleanup test for Safari<br> <li> Import assertion utilities<br> <li> Use invalid capability to cause failure<br> <li> Test service stops when session creation fails

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16038/files#diff-c0bb73b3c1e76a971e4877983cdf5de92349483293cc37a472641493c7c2146e">+21/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

File: java/src/org/openqa/selenium/remote/RemoteWebDriver.java [hunk 1]
```diff
@@ -241,46 +242,58 @@ protected void startSession(Capabilities capabilities) {
...
+    try {
+      Response response = execute(DriverCommand.NEW_SESSION(singleton(capabilities)));
 
-    if (response == null) {
-      throw new SessionNotCreatedException(
-          "The underlying command executor returned a null response.");
-    }
+      if (response == null) {
+        throw new SessionNotCreatedException(
+            "The underlying command executor returned a null response.");
```
Comment (pujagani) [inline java/src/org/openqa/selenium/remote/RemoteWebDriver.java:250]: Please avoid formatting changes as part of this PR. Please create another PR if formatting changes are required.

Comment (iampopovich) [reply java/src/org/openqa/selenium/remote/RemoteWebDriver.java:250]: my bad , it seems i did not run scripts/format.sh

Comment (iampopovich) [reply java/src/org/openqa/selenium/remote/RemoteWebDriver.java:250]: applied formatting with format.sh script . @pujagani thanks !

**Comment by pujagani** [comment]
The changes look good to me. Will wait for CI to pass before merging it.

**Review by pujagani — APPROVED** [review]
Thank you! @iampopovich
